### PR TITLE
Refactor zinc to use split_common 

### PR DESCRIPTION
**Pull request by monksoffunk**
This removes own split related codes and uses split_common files.

## Description

This removes following files under the zinc/revx/:

- matrix.c

- serial_config.h

- serial_config_simpleapi.h

- split_scomm.h

- split_scomm.c

- split_util.c

- split_util.h

Also, new RGB LED options added:

- Backlight or/and underglow feature

- Now can select from continuous RGB Animations between LR or separate RGB Animations (the same as before)
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
As mentioned in #7111 you have modified user keymaps, therefore i am adding the breaking change. 

Also there are a few tidy up changes in #7111 that I would want applied to this PR, I will add them as a review shortly, though its a shame we now have 2 PRs the main thing is we are now targeting split_common.

**Comment by monksoffunk** [comment]
@zvecr I tried to review #7111, but too many things to add and I've already prepared this PR, so I choose open this. I'm sorry about that.
Anyway thanks.

**Comment by monksoffunk** [comment]
Why is keyboards/zinc/serial.c listed on conflicting files though it is deleted?

**Comment by fauxpark** [comment]
@monksoffunk that's due to [this commit](https://github.com/qmk/qmk_firmware/commit/363cdb5fc00519a4eb45a4401b93451949c59d6a#diff-9df33efec6f0539d9d5c4b6319732aae). The contents of the deleted serial.c are not the same as the one in master.

You should be able to `git merge master` from this branch, then `git mergetool` to "keep" the deleted file, and finally `git merge --continue` to commit.

**Comment by monksoffunk** [comment]
@fauxpark Thanks
@zvecr  What about your review?

**Comment by Duckle29** [comment]
@monksoffunk You would perform those merges on your own branch.

First of all make sure your master branch is up to date with upstream, then do what fauxpark suggested on your zinc_splitcommon_2019OCT branch :)

File: keyboards/zinc/readme.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Zinc
 40% row-staggered split keyboard.
 
 Keyboard Maintainer: [monksoffunk](https://github.com/monksoffunk/) [@monksoffunkJP](https://twitter.com/monksoffunkJP)
```
Comment (fauxpark) [inline keyboards/zinc/readme.md]:
```suggestion
* Keyboard Maintainer: [monksoffunk](https://github.com/monksoffunk/) [@monksoffunkJP](https://twitter.com/monksoffunkJP)
```

File: keyboards/zinc/readme.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Zinc
 40% row-staggered split keyboard.
 
 Keyboard Maintainer: [monksoffunk](https://github.com/monksoffunk/) [@monksoffunkJP](https://twitter.com/monksoffunkJP)  
-Hardware Supported: Zinc PCB  
+Hardware Supported: Zinc PCB
```
Comment (fauxpark) [inline keyboards/zinc/readme.md]:
```suggestion
* Hardware Supported: Zinc PCB
```

File: keyboards/zinc/readme.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ Zinc
...
 Hardware Availability: (https://twitter.com/monksoffunkJP)
```
Comment (fauxpark) [inline keyboards/zinc/readme.md]:
```suggestion
* Hardware Availability: (https://twitter.com/monksoffunkJP)
```

File: keyboards/zinc/rules.mk [hunk 4]
```diff
@@ -28,11 +28,11 @@ UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
 SUBPROJECT_rev1 = no
-USE_I2C = yes
+USE_I2C = no
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
 
-CUSTOM_MATRIX = yes
+CUSTOM_MATRIX = no
```
Comment (fauxpark) [inline keyboards/zinc/rules.mk]:
```suggestion
```

File: keyboards/zinc/rules.mk [hunk 5]
```diff
@@ -28,11 +28,11 @@ UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
 SUBPROJECT_rev1 = no
-USE_I2C = yes
+USE_I2C = no
```
Comment (fauxpark) [inline keyboards/zinc/rules.mk]:
```suggestion
```

File: keyboards/zinc/keymaps/toshi0383/rules.mk [hunk 6]
```diff
@@ -1,11 +1,12 @@
...
-#   change yes to no to disable
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
 #
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = no        # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/toshi0383/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/zinc/keymaps/toshi0383/rules.mk [hunk 7]
```diff
@@ -1,11 +1,12 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/toshi0383/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no       # Mouse keys
```

File: keyboards/zinc/keymaps/toshi0383/rules.mk [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/toshi0383/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/zinc/keymaps/toshi0383/rules.mk [hunk 9]
```diff
@@ -1,11 +1,12 @@
...
+CONSOLE_ENABLE = no         # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/toshi0383/rules.mk]:
```suggestion
CONSOLE_ENABLE = no         # Console for debug
```

File: keyboards/zinc/keymaps/toshi0383/rules.mk (hunk 6 above)
Comment (monksoffunk) [reply keyboards/zinc/keymaps/toshi0383/rules.mk]: I checked the latest version of qmk, so many '# Virtual DIP switch configuration(+1000)' exist...
864 results in 864 files 
Should we change?

Comment (fauxpark) [reply keyboards/zinc/keymaps/toshi0383/rules.mk]: Yes, see https://github.com/qmk/qmk_firmware/pull/6947

**Comment by zvecr** [comment]
PR still modifies the @ginjake keymap, we either need sign off or go through breaking changes.

**Comment by monksoffunk** [comment]
@zvecr thank you. I just add commits as you did #7111.

> PR still modifies the @ginjake and @toshi0383 keymap, we either need sign off or go through breaking changes.
Specifically which codes?

**Comment by Curry** [comment]
@monksoffunk  with the addition of #7915, You need to add
`#define DIODE_DIRECTION COL2ROW` to the `config.h` files, otherwise all keymap builds will fail.

File: keyboards/zinc/rev1/config.h [hunk 10]
```diff
@@ -45,19 +40,13 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 /* key matrix size */
 // Rows are doubled-up
-  #define MATRIX_ROWS 8
-  #define MATRIX_ROW_PINS { F6, F7, B1, B3 }
+#define MATRIX_ROWS 8
+#define MATRIX_ROW_PINS { F6, F7, B1, B3 }
 
 // wiring of each half
 #define MATRIX_COLS 6
 #define MATRIX_COL_PINS { F4, D4, C6, D7, E6, B4 }
```
Comment (Curry) [inline keyboards/zinc/rev1/config.h]:
```suggestion

#define DIODE_DIRECTION COL2ROW

```

File: keyboards/zinc/reva/config.h [hunk 11]
```diff
@@ -42,19 +40,13 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
 /* key matrix size */
 // Rows are doubled-up
-  #define MATRIX_ROWS 8
-  #define MATRIX_ROW_PINS { D4, C6, D7, E6 }
+#define MATRIX_ROWS 8
+#define MATRIX_ROW_PINS { D4, C6, D7, E6 }
 
 // wiring of each half
 #define MATRIX_COLS 6
 #define MATRIX_COL_PINS { F4, F5, F6, F7, B1, B3}
```
Comment (Curry) [inline keyboards/zinc/reva/config.h]:
```suggestion

#define DIODE_DIRECTION COL2ROW

```

**Comment by monksoffunk** [comment]
> @monksoffunk with the addition of #7915, You need to add
> `#define DIODE_DIRECTION COL2ROW` to the `config.h` files, otherwise all keymap builds will fail.

@Curry Thanks!

**Comment by monksoffunk** [comment]
@toshi0383 @ginjake 
I updated your keymaps at this PR.
Please confirm that it works properly.

**Comment by drashna** [comment]
@toshi0383 to make sure, you flashed to both sides of the keyboard, right?   If not, because this is a significant change to the communication code, it won't work properly.

**Comment by ginjake** [comment]
@monksoffunk 
thanks updated my keymaps.
looks good to me.

**Comment by monksoffunk** [comment]
@ginjake  @toshi0383 
Thanks!

@zvecr 
2 keymaps that I modified is now confirmed.

File: keyboards/zinc/keymaps/default/keymap.c [hunk 12]
```diff
@@ -140,21 +141,21 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { \
...
    * |------+------+------+------+------+------|             |------+------+------+------+------+------|
    * |      |      |      | EISU | EISU | EISU |             | KANA | KANA | Home |PageDn|PageUp| End  |
    * `-----------------------------------------'             `-----------------------------------------'
    */
     [_ADJUST] =  LAYOUT_ortho_4x12( \
       _______, RESET,   RGBRST,  _______, _______, _______,                   _______, QWERTY,  COLEMAK, DVORAK,  _______, KC_INS, \
       _______, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, AG_NORM,                   AG_SWAP, KC_MINS, KC_EQL,  KC_PSCR, KC_SLCK, KC_PAUS,\
-      _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,                   _______, _______, _______, _______, _______, _______,\
-      _______, _______, _______, EISU,    EISU,    EISU,                      KANA,    KANA,    KC_HOME, KC_PGDN, KC_PGUP, KC_END\
+      RGBMODR, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,                   _______, _______, _______, _______, KC_PGUP, _______,\
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/default/keymap.c]:
```suggestion
      RGB_RMOD,RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______,                   _______, _______, _______, _______, KC_PGUP, _______,\
```

File: keyboards/zinc/keymaps/default/keymap.c [hunk 13]
```diff
@@ -245,16 +246,28 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
         }
         return false;
         break;
+
       //led operations - RGB mode change now updates the RGB_current_mode to allow the right RGB mode to be set after reactive keys are released
+    case RGBMODR:
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/default/keymap.c]:
```suggestion
    case RGB_RMOD:
```

File: keyboards/zinc/keymaps/default/keymap.c [hunk 14]
```diff
@@ -33,6 +33,7 @@ enum custom_keycodes {
   BACKLIT,
   KANA,
   EISU,
+  RGBMODR,
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/default/keymap.c]:
```suggestion
```

File: keyboards/zinc/keymaps/default/rules.mk [hunk 15]
```diff
@@ -20,13 +20,20 @@ define ZINC_CUSTOMISE_MSG
   $(info Zinc customize)
   $(info -  LED_BACK_ENABLE=$(LED_BACK_ENABLE))
   $(info -  LED_UNDERGLOW_ENABLE=$(LED_UNDERGLOW_ENABLE))
+  $(info -  LED_BOTH_ENABLE=$(LED_BOTH_ENABLE))
+  $(info -  LED_RGB_CONT=$(LED_RGB_CONT))
+  $(info -  RGB_MATGRIX=$(RGB_MATRIX))
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/default/rules.mk]:
```suggestion
  $(info -  RGB_MATRIX=$(RGB_MATRIX))
```

File: keyboards/zinc/keymaps/default/keymap.c (hunk 12 above)
Comment (monksoffunk) [reply keyboards/zinc/keymaps/default/keymap.c]: Thanks for your changes.

File: keyboards/zinc/keymaps/default/readme_en.md [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+```
+$ cd qmk_firmware
+```
+make with `zinc:<keymap_name>`
+
+```
+$ make zinc:default
+```
+
+To make and flash ProMicro with `:avrdude`
```
Comment (Duckle29) [inline keyboards/zinc/keymaps/default/readme_en.md]:
```suggestion
To make and flash with `:flash`
```

File: keyboards/zinc/keymaps/default/readme_en.md [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+
+```
+$ make zinc:default:avrdude
```
Comment (Duckle29) [inline keyboards/zinc/keymaps/default/readme_en.md]:
```suggestion
$ make zinc:default:flash
```

**Comment by monksoffunk** [comment]
@zvecr @drashna 
I believe I've done what is expected.
The authors of keymaps have confirmed this PR.
Is there anything else I should do?

**Comment by monksoffunk** [comment]
@zvecr @Duckle29 
Re-review please!

**Comment by monksoffunk** [comment]
@noroadsleft 
Thanks for your help!
I want to make sure the location of ChangeLog file that I need to create.
You mean (/zinc/docs/ChangeLog/20200530/PR7114.md)?

**Comment by noroadsleft** [comment]
> I want to make sure the location of ChangeLog file that I need to create.
> You mean (qmk_firmware/keyboards/zinc/docs/ChangeLog/20200530/PR7114.md)?

ChangeLog should be at `qmk_firmware/docs/ChangeLog/20200530/PR7114.md`.

**Comment by noroadsleft** [comment]
@monksoffunk I may want to flesh out the changelog entry some more; probably in #8978. Would appreciate your input there should I mention you on that PR.

Code-wise I'm calling this good.

Thanks!

**Pull request merged**